applications/dashboard/controllers/class.utilitycontroller.php

```diff
@@ -38,8 +68,16 @@ class UtilityController extends DashboardController {
 
     /**
      * UtilityController constructor.
+     *
+     * @param ConfigurationInterface $configuration
+     * @param EventManager $eventManager
+     * @param SessionInterface $session
      */
-    public function __construct() {
+    public function __construct(ConfigurationInterface $configuration, EventManager $eventManager, SessionInterface $session) {
+        $this->configuration = $configuration;
+        $this->eventManager = $eventManager;
+        $this->session = $session;
+
         parent::__construct();
```

Comment:
Is there any specific reason why we need to extend UtilityController instead of create new one?

Reply:
I noticed the same thing, but it was part of the acceptance criteria that we had initially discussed.

Reply:
It looks like "bad practice in action" to me.

Reply:
Only that we've planned it poorly again. I think the purpose was to push this quickly to satisfy the ops team.

If we were to really make a small project out of it, I would have advocated for making this directly using `Vanilla\Web\Page`.

Reply:
> Any reason to advocate that decision?

I believe the original reasoning was partially that this migration of functionality from Sitemaps to core should not include breaking changes. If we put it in its own controller (i.e. move it to a different URL), then the /utility/robots URL from Sitemaps would be broken.

Reply:
But isn't it possible to have any controller/method bind to the route here:
```
Gdn::router()->setRoute("robots.txt", "/utility/robots", "Internal");
```
?

Reply:
It's definitely possible to route robots.txt to any other URL. But that doesn't change the fact that /utility/robots was a functioning URL in Sitemaps that will no longer be available.

Reply:
```
Gdn::router()->setRoute("robots.txt", "/robots/general", "Internal");
Gdn::router()->setRoute("utility/robots", "/robots/general", "Internal");
```
kind?